nestjs/src/courses/score/score.service.ts

```diff
@@ -0,0 +1,151 @@
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import * as _ from 'lodash';
+
+import { Student } from '@entities/student';
+
+import { ScoreTableFilters } from '@common/types/score';
+
+import { paginate } from 'src/lib/paginate';
+import { MentorsService } from 'src/courses/mentors';
+import { getContactsUserFields, getPrimaryUserFields } from 'src/core/helpers';
+
+import { orderByFieldMapping, OrderDirection, OrderField } from './dto/GetScoreQuery.dto';
+import { InterviewsService } from '../interviews';
+import { ScoreDto, ScoreStudentDto } from './dto/Score.dto';
+
+const defaultFilter: ScoreTableFilters = {
+  activeOnly: false,
+  githubId: '',
+  name: '',
+  'mentor.githubId': '',
+  cityName: '',
+};
+
+const defaultOrder: { field: OrderField; direction: OrderDirection } = {
+  field: 'rank',
+  direction: 'ASC',
+};
+
+type ScoreOptions = {
+  includeContacts?: boolean;
+};
+
+@Injectable()
+export class ScoreService {
+  constructor(
+    private courseId: number,
```

Comment:
consider making the service stateless, ie passing courseid for respective methods